library/general/src/modules/ProductEvaluation.rb

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2packager/medium_type"
+require "y2packager/resolvable"
+require "y2packager/repository"
+
+module Yast
+  class ProductEvaluationClass < Module
```

Comment:
IMHO this class does too much, it should just provide the basic framework for collecting the data and dumping it to a file.

The rest should be implemented outside, in the other modules. E.g. the registration module should register a callback to dump the addons. This code should log just the basic data (like upgrade/new installation), leave the details to other modules.

If we change something in registration we will need to adapt also this place. And we will very likely forget about that... :thinking: 

And I'd also rename the class to something more generic, like `InstallationInfo`, `InstallationDebug`. Also the file should be save to `/var/log/YaST2/installation_info` or something like that.

We can log any details we need, it might not be just about the products (although it's the main purpose).

Reply:
Well, the benefit of this is that we do not have to adapt so many packages and provide at least a basic information which of course can be improved by other modules too. Not sure.....